backend/api_calls.py

```diff
@@ -261,17 +261,21 @@ def AddGroup(request, firebase):
   """
   results = []
   valid_args = ['!groupId', 'gameId', 'ownerPlayerId', 'allegianceFilter']
-  required_args = list(valid_args)
+  required_args = ['groupId', 'gameId', 'allegianceFilter']
   required_args.extend(['autoAdd', 'autoRemove', 'membersCanAdd', 'membersCanRemove'])
   ValidateInputs(request, firebase, required_args, valid_args)
 
-  put_args = list(required_args)
-  put_args.remove('!groupId')
+  put_args = set(required_args) - set(['groupId'])
   group_data = {k: request[k] for k in put_args}
```

Comment:
That could include extraneous args that the user may pass in. We're not checking for additional un-required args.